src/main.ts

```diff
@@ -11,33 +11,27 @@ async function run(): Promise<void> {
 
     const config: Configuration = JSON.parse(fs.readFileSync(`${path}/openapi-merge.json`, 'utf-8'));
 
-    const urls = config.inputs.map(({ inputFile }) => inputFile);
-
-    for (const [index, url] of urls.entries()) {
-      const cleanUrl = url.replace('https://github.com/', '').replace('blob/', '');
-      const rawUrl = `https://raw.githubusercontent.com/${cleanUrl}`;
-
-      const { data } = await axios.get(rawUrl, {
-        headers: {
-          Authorization: `Bearer ${token}`,
-          'Content-Type': 'application/json',
-          Accept: '*/*',
-        },
-      });
-
-      fs.writeFileSync(`${path}/openapi-${index}.yaml`, data);
-
-      config.inputs[index].inputFile = `openapi-${index}.yaml`;
-    }
-
-    fs.writeFileSync(`${path}/openapi-merge.json`, JSON.stringify(config, null, 2));
-
-    execSync(`npx openapi-merge-cli --config ${path}/openapi-merge.json`);
-
-    execSync('git clean -f');
-
-    execSync(`git restore ${path}/openapi-merge.json`);
+    // array of yaml
+    const yamlData = await Promise.all(
+      config.inputs.map(async ({ inputFile }) => {
+        const cleanUrl = inputFile.replace('https://github.com/', '').replace('blob/', '');
+        const rawUrl = `https://raw.githubusercontent.com/${cleanUrl}`;
+
+        const { data } = await axios.get(rawUrl, {
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            Accept: '*/*',
+          },
+        });
+
+        return yaml.load(data);
+      })
+    );
+    const jsonData = JSON.stringify(yamlData);
+    fs.writeFileSync(path + config.output, jsonData);
   } catch (error) {
+    console.log('error', error);
```

Comment:
I might be wrong but I think if you want to print out an error in the GH workflow context, it has to be a method under the `core.`, `console.log()` won't be working. But I might be wrong.

Reply:
ah I left both actually, the consol.log is for debugging, I could remove it if you like